context/src/test/java/io/grpc/ContextTest.java

```diff
@@ -763,6 +765,22 @@ public void initContextWithCustomClassLoaderWithCustomLogger() throws Exception
     ((Runnable) runnable.getDeclaredConstructor().newInstance()).run();
   }
 
+  /**
+   * The context storage has per thread data structures.
+   * Ensure that newly created threads can attach/detach.
+   */
+  @Test
+  public void attachDetachNewThread() throws ExecutionException, InterruptedException {
+    Future<?> future = scheduler.submit(Context.current().wrap(new Runnable() {
```

Comment:
Good suggestion on the current value, I will update the tests. As for per thread data structures, I was previously just referring to initializing the ThreadLocal to ROOT, which is not necessary anymore with ejona's suggestions.